src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -100,7 +100,7 @@ public CachingContext(JsonSerializerOptions options)
             // Property only accessed by reflection in testing -- do not remove.
             // If changing please ensure that src/ILLink.Descriptors.LibraryBuild.xml is up-to-date.
             public int Count => _converterCache.Count + _jsonTypeInfoCache.Count;
-            public JsonConverter GetOrAddConverter(Type type) => _converterCache.GetOrAdd(type, Options.GetConverterFromType);
+            public JsonConverter GetOrAddConverter(Type type) => _converterCache.GetOrAdd(type, s_converterRetrievalFunc!(type, Options));
```

Comment:
Is this always going to call the `s_converterRetrievalFunc` even if the `type` is already cached?